File: practice/problem1/ans.py
class Solution:

    def twoSum(self, nums: List[int], target: int) -> List[int]:

        num_map = {}

        for idx, num in enumerate(nums):

            if (target - num) in num_map:

                return [idx, num_map[target - num]]

            else:

                num_map[num] = idx

        

        

File: practice/problem1/ans.cpp
class Solution {

public:

    vector<int> twoSum(vector<int>& nums, int target) {

        unordered_map<int, int> num_map;

        auto num_idx = 0;

        for(auto num: nums) {

            int candidate_num = target - num;

            if(num_map.find(candidate_num) != num_map.end()) {

                return vector<int> {num_idx, num_map[candidate_num]};

            } 

            

            num_map[num] = num_idx;

            num_idx++; 

        }

        

        return vector<int> {};

    }

};
